tensordict/prototype/tensorclass.py

```diff
@@ -27,6 +27,23 @@
 UNION_PATTERN = re.compile(r"Union\[(.*?)\]")
 
 
+class InvalidAttributeName(ValueError):
+    pass
```

Comment:
This might be overkill, but I couldn't see a built-in exception that accurately described the problem of an invalid attribute name. I think `ValueError` is reasonable, because naming attributes in a class definition is equivalent to passing those names as `attrs` argument to some metaclass, but to the user it feels quite different to calling a function with arguments.

Reply:
Have you thought about AttributeError (even if it's semantically quite the opposite)?

Reply:
e.g.
```
>>> class A:
...     pass
...
>>> A.__dict__ = 1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: attribute '__dict__' of 'type' objects is not writable
```

Reply:
Yes, `AttributeError` was the main alternative I considered, though as you say it's more about accessing something that doesn't exist rather than writing over something that does.

Happy to change it if you think that would be better? Even if it's not the conventional usage of AttributeError, it is an error relating to attributes...

And what about the custom exception? Just us a bare AttributeError instead?